### PR TITLE
Add navigator type definition

### PR DESCRIPTION
**Pull request by ruihe774**
### What does this PR do?

This adds TypeScript type for `navigator`.

- [x] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: packages/bun-types/globals.d.ts [hunk 1]
```diff
@@ -3800,3 +3800,11 @@ interface PromiseConstructor {
     reject: (reason?: any) => void;
   };
 }
+
+interface Navigator {
+  readonly userAgent: string;
+  readonly platform: "MacIntel" | "Win32" | "Linux x86_64",
+  readonly hardwareConcurrency: number
```
Comment (ghiscoding) [inline packages/bun-types/globals.d.ts]: I think both of these lines should end with semicolon `;` not comma or empty

Comment (ruihe774) [reply packages/bun-types/globals.d.ts]: Fixed

**Comment by ghiscoding** [comment]
looks good now :)

**Comment by ruihe774** [comment]
@ghiscoding Would you approve it?

**Comment by ghiscoding** [comment]
No sorry I do not have any access to this repo, a maintainer would need to approve. I'm just a regular user that saw some improvement on your original commit. Maybe @paperdave is a good person to review

**Review by paperclover — APPROVED** [review]

**Pull request merged**